### PR TITLE
Pings: speed up growth query

### PR DESCRIPTION
**Pull request by camdencheek**
This speeds up a query on the event logs table that was taking 30 seconds on sourcegraph.com so now it only takes 1 second. 

## Test plan

Existing tests, ran manually on sourcegraph.com to ensure I got the same numbers.

### CODE REVIEW

File: internal/usagestats/growth.go [hunk 1]
```diff
@@ -42,25 +42,28 @@ type usersGrowthStatistics struct {
 
 func getUsersGrowthStatistics(ctx context.Context, db database.DB) (*usersGrowthStatistics, error) {
 	const usersQuery = `
-WITH all_usage_by_user_and_month AS (
-    SELECT user_id,
-           DATE_TRUNC('month', timestamp) AS month_active
-      FROM event_logs
-     GROUP BY user_id,
-              month_active),
```
Comment (camdencheek) [inline internal/usagestats/growth.go:50]: We don't need to analyze all of history. Only the last two months matter. Additionally, we can avoid doing the DATE_TRUNC calculation on every row in the database which is required in order to group by `month_active`.

File: internal/usagestats/growth.go [hunk 2]
```diff
@@ -78,14 +81,6 @@ SELECT COUNT(*) FILTER ( WHERE recent_usage_by_user.created_month = DATE_TRUNC('
...
   FROM recent_usage_by_user
     `
-	const accessRequestsQuery = `
-	SELECT
-		COUNT(*) FILTER (WHERE status = 'PENDING') AS pending_access_requests,
-		COUNT(*) FILTER (WHERE status = 'APPROVED') AS approved_access_requests,
-		COUNT(*) FILTER (WHERE status = 'REJECTED') AS rejected_access_requests
-	FROM access_requests
-	WHERE created_at >= DATE_TRUNC('month', $1::timestamp)
-	`
```
Comment (camdencheek) [inline internal/usagestats/growth.go:88]: this was unused

File: internal/usagestats/growth.go [hunk 3]
```diff
@@ -42,23 +42,26 @@ type usersGrowthStatistics struct {
...
+    SELECT DISTINCT user_id
+    FROM event_logs
+    WHERE timestamp > (DATE_TRUNC('month', $1::timestamp) - INTERVAL '1 month')
+        AND timestamp < DATE_TRUNC('month', $1::timestamp)
+),
+active_this_month AS (
+    SELECT DISTINCT user_id
+    FROM event_logs
+    WHERE timestamp > DATE_TRUNC('month', $1::timestamp)
+        AND timestamp < (DATE_TRUNC('month', $1::timestamp) + INTERVAL '1 month')
```
Comment (eseliger) [inline internal/usagestats/growth.go:55]: this means "the month of today, so 5/1, plus a month so 6/1" to get any time in the current month, right?

Comment (camdencheek) [reply internal/usagestats/growth.go:55]: Yep, exactly

**Pull request merged**